src/Symfony/Component/Intl/Locale.php

```diff
@@ -69,6 +69,11 @@ public static function getFallback(string $locale): ?string
     {
         if (\function_exists('locale_parse')) {
             $localeSubTags = locale_parse($locale);
+
+            if (null === $localeSubTags) {
```

Comment:
> Returns null when the length of locale exceeds INTL_MAX_LOCALE_LEN.

however the actual return type is documented to be array, so i think this is an error case actually. Thus a throw.

returning null might imply eg. `root` which has no parent

Reply:
@ro0NL thanks for your review. Sorry but I'm a little confused. What is your recommendation for this issue?

Reply:
something like

```
if (null === $localeSubTags = locale_parse($locale)) {
   throw new InvalidLocaleException();
}
```

Not sure we have such exception, otherwise generic \LogicException or \InvalidArgumentException fits as well.

getFallback('root') // null
getFallbck(str_repeat('x', 1000)) // throw